### ISSUE TITLE
Help: Type mismatch: inferred type is PluginRegistry but FlutterEngine was expected

### ISSUE THREAD

**Issue by januwA** [fluttercommunity/flutter_downloader#205]
I got this error while building:
```
C:\Users\ajanuw\Desktop\flutter_video_app\android\app\src\main\kotlin\com\ajanuw\flutter_video_app\MyApplication.kt: (10, 48): Type mismatch: inferred type is PluginRegistry but FlutterEngine was expected
```

MyApplication.kt
```
package com.ajanuw.flutter_video_app

import io.flutter.app.FlutterApplication
import io.flutter.plugin.common.PluginRegistry
import io.flutter.plugins.GeneratedPluginRegistrant
import vn.hunghd.flutterdownloader.FlutterDownloaderPlugin

internal class MyApplication : FlutterApplication(), PluginRegistry.PluginRegistrantCallback {
    override fun registerWith(registry: PluginRegistry) {
        GeneratedPluginRegistrant.registerWith(registry)
    }
}
```

```
λ flutter doctor -v
[√] Flutter (Channel master, v1.13.6-pre.14, on Microsoft Windows [Version 10.0.17763.914], locale zh-CN)
    • Flutter version 1.13.6-pre.14 at D:\flutter
    • Framework revision 924efae96c (3 days ago), 2019-12-20 20:36:29 -0800
    • Engine revision 33813929e3
    • Dart version 2.8.0 (build 2.8.0-dev.0.0 886615d0f9)

[!] Android toolchain - develop for Android devices (Android SDK version 29.0.0-rc3)
    • Android SDK at D:\Android\SDK
    • Android NDK location not configured (optional; useful for native profiling support)
    • Platform android-29, build-tools 29.0.0-rc3
    • ANDROID_HOME = D:\Android\SDK
    • Java binary at: D:\Android\Android_Studio\jre\bin\java
    • Java version OpenJDK Runtime Environment (build 1.8.0_202-release-1483-b03)
    X Android license status unknown.
      Try re-installing or updating your Android SDK Manager.
      See https://developer.android.com/studio/#downloads or visit https://flutter.dev/setup/#android-setup
      for detailed instructions.

[√] Chrome - develop for the web
    • Chrome at C:\Program Files (x86)\Google\Chrome\Application\chrome.exe

[√] Android Studio (version 3.5)
    • Android Studio at D:\Android\Android_Studio
    • Flutter plugin version 42.1.1
    • Dart plugin version 191.8593
    • Java version OpenJDK Runtime Environment (build 1.8.0_202-release-1483-b03)

[√] Connected device (3 available)
    • SM G8750   • c5976a52   • android-arm64  • Android 9 (API 28)
    • Chrome     • chrome     • web-javascript • Google Chrome 78.0.3904.108
    • Web Server • web-server • web-javascript • Flutter Tools

! Doctor found issues in 1 category.
```

**Comment by hnvn** [comment]
It is broken since Flutter v1.12.13, I am working on #198 to fix it.

**Comment by redabelca** [comment]
i have that same issue .
Flutter really become bothering me with these huge changes that break all the flutter plugins made in this world suddenly.

**Comment by hnvn** [comment]
See https://github.com/fluttercommunity/flutter_downloader/issues/214#issuecomment-570776556

**Comment by hnvn** [comment]
fixed in v1.4.0
